# **[NGS_data_analysis_note](https://github.com/Zheyu-Ding/NGS_data_analysis_note/)**
## **Background**
The three tumultuous years of my master's program are coming to an end, and the result of my PhD application remains unknown. <br> 
Here, I would like to organize the next-generation sequencing(NGS) data analysis I have done during my master's phase, mainly focusing on learning materials and scripts. <br>
The construction of this repository is based on the [bioinformatics notes](https://github.com/Zheyu-Ding/NGS-data-analysis) I initially reviewed, which were provided by [Dr.Fajin Li](https://scholar.google.com/citations?hl=en&user=Vj3JwOkAAAAJ). <br>
I am extremely fortunate to have met him during my master's studies. Under his guidance, I began my journey into bioinformatics, not only resolving doubts in my research topics but also gradually understanding how to think critically in scientific research and how to solve the problems within the project. More importantly, I have learned valuable qualities of a outstanding researcher from him.

## **Design**
So far, the data analysis I have been involved in has primarily focused on translation regulation using ribosome profiling technology. Additionally, I have also completed data analysis for RNA-seq, scRNA-seq, and PAR-CLIP seq, but these analyses were not particularly in-depth.

## **Project**
1. [Ribo-seq data analysis](https://github.com/Zheyu-Ding/NGS_data_analysis_note/blob/main/Projects/Ribo-seq.md)
2. RNA-seq data analysis
3. PAR-CLIP seq data analysis
4. scRNA-seq data analysis